### ISSUE TITLE
Long sheet names + chop off cause merge failure

### ISSUE THREAD

**Issue by jake-stack** [saga-vcs/saga-excel#83]
In the following template: [Template-Cap-Table-Alexander-Jarvis.xlsx](https://github.com/saga-vcs/saga-excel/files/4686489/Template-Cap-Table-Alexander-Jarvis.xlsx)

Just creating a project and merging causes an error. After a bit of investigation, this appears to be a result of the prefix we append to the sheet names, which makes them too long to be valid excel sheet names. 

In turn, this causes an error in line 40 of the commit function which occurs at the start of merge, when we try and rename the newly inserted sheets to their to-long names.

**Comment by naterush** [comment]
Fixes I can think of: 
- Report an error screen that tells ppl if they sheet names are too long (temp fix for now)
- Store a bijection from sheet names <-> short sheet names (where we store them in the commit). The template for this can be `saga-<commit_id>-<short_name>`, and then we convert them fowards when we are commiting, and backwards when we are resetting personal or merging.

The problem with the second option is that we need to keep this data in a sheet somewhere (and not just in app memory), else we will lose it when we try and go back. Which is NBD, but seems like it might be kinda annoying. 

The other thing is that would be nice is to a) not maintain a bunch of extra shit for each merge, and b) have stability in the mapping across merges (e.g. sheet1 -> s1 in commit 1, then sheet1 -> s1 in later merges). This seems non-trivial to do in a clean way.

Also, while we're at-it, we might as well fix the bug where a sheet w/ the prefix of another sheet might get screwed up. Also, the bug where I think if there is a Sheet1 and Sheet1 (2) already in the project, and you merge, one of them may be deleted.

TODO: add a test for commit + merge w/ long sheet names, to make sure we fix things.

**Comment by naterush** [comment]
@aarondr77 please let me know what you think - or if you can think of other options!

**Comment by jake-stack** [comment]
I shortened the names of the tabs but the merge took to long that it froze, probably because the file has 42 tabs and it thus too large

**Comment by aarondr77** [comment]
Yeah, the best thing I can think of right now is just hashing the sheet names and storing a mapping from hashed sheet name to original sheet name. We could hash the given sheet name with the commit ID to ensure uniqueness.

**Comment by naterush** [comment]
@aarondr77 Gonna start by writing some tests that catch these bugs as they currently exist.
